src/Service/PromotionsService.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\PromotionRepository;
+use BBC\ProgrammesPagesService\Domain\ApplicationTime;
+use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\PromotionMapper;
+
+class PromotionsService extends AbstractService
+{
+    /** @var PromotionRepository */
+    protected $repository;
+
+    public function __construct(
+        PromotionRepository $repository,
+        PromotionMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    /**
+     * @return Promotion[]
+     */
+    public function findActivePromotionsByEntity(
+        CoreEntity $coreEntity,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE,
+        $ttl = CacheInterface::NORMAL
+    ): array {
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, $coreEntity->getPid(), $limit, $page, $ttl);
+
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () use ($coreEntity, $limit, $page) {
+                $dbEntities = $this->repository->findActivePromotionsByContext(
+                    $coreEntity->getDbId(),
+                    ApplicationTime::getTime(),
+                    $limit,
+                    $this->getOffset($limit, $page)
+                );
+
+                return $this->mapManyEntities($dbEntities);
+            }
+        );
+    }
+
+    /**
+     * @return Promotion[]
+     */
+    public function findActiveSuperPromotionsByEntity(
+        CoreEntity $coreEntity,
```

Comment:
no. Actually is a little confuse. Maybe a better name should be ContextEntity. The promotable one is the thing that is being promoted. The context is "who" is promoting it. Context is maybe better